tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -54,29 +58,96 @@ struct SynthOrdered {
     }
 };
 
-template <class T>
-inline constexpr bool is_pair = false;
-template <class A, class B>
-inline constexpr bool is_pair<std::pair<A, B>> = true; // TRANSITION, std::pair spaceship not yet implemented
+struct OrderedChar {
+    OrderedChar() = default;
+    OrderedChar(const char c) : c(c) {}
 
-template <class Container>
-void ordered_containers_test(const Container& smaller, const Container& smaller_equal, const Container& larger) {
+    OrderedChar& operator=(const char& other) {
+        c = other;
+        return *this;
+    }
+
+    auto operator<=>(const OrderedChar&) const = default;
+
+    operator char() const {
+        return c;
+    }
+
+    char c;
+};
+
+struct WeaklyOrderedChar : OrderedChar {};
+struct WeaklyOrderdByOmissionChar : OrderedChar {};
+struct PartiallyOrderedChar : OrderedChar {};
+
+namespace std {
+    template <>
+    struct char_traits<OrderedChar> : public char_traits<char> {
+        using char_type = OrderedChar;
+
+        static int compare(const char_type* first1, const char_type* first2, size_t count) {
+            for (; 0 < count; --count, ++first1, ++first2) {
+                if (*first1 != *first2) {
+                    return *first1 < *first2 ? -1 : +1;
+                }
+            }
+
+            return 0;
+        }
+
+        static bool eq(const char_type l, const char_type r) {
+            return l.c == r.c;
+        }
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderedChar> : public char_traits<OrderedChar> {
+        using char_type           = WeaklyOrderedChar;
+        using comparison_category = weak_ordering;
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderdByOmissionChar> : public char_traits<OrderedChar> {
+        using char_type = WeaklyOrderdByOmissionChar;
+    };

```

Comment:
I am confused about what's happening here. My understanding is:
* `char_traits<char>` should define `comparison_category` to be `strong_ordering`, but doesn't yet.
* `char_traits<OrderedChar>` derives from `char_traits<char>`, so it'll pick up `comparison_category` as `strong_ordering` in the future, but not yet.
* `char_traits<WeaklyOrderdByOmissionChar>` derives from `char_traits<OrderedChar>`, so it'll pick up `comparison_category` as `strong_ordering` in the future, but not yet.

However, the "By Omission" seems to indicate that the intent is for this specialization to forever lack a `comparison_category` typedef, so as to activate the `weak_ordering` default.

Perhaps this should define a *private* typedef for `strong_ordering`, which will be inaccessible and hence treated as missing, activating the `weak_ordering` default? (It's okay that this is a `struct`.)

Reply:
My intention was to validate the default behavior like you pointed out. I forgot that I was inheriting from `char_traits<char>` and would get the `comparison_category` from that in the future.